codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/ClientClassUtils.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.codegen.poet.client;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import java.util.Optional;
+import java.util.function.Function;
+import software.amazon.awssdk.codegen.model.intermediate.OperationModel;
+import software.amazon.awssdk.codegen.model.intermediate.ShapeModel;
+import software.amazon.awssdk.http.HttpResponseHandler;
+
+public final class ClientClassUtils {
+    private ClientClassUtils() {
+    }
+
+    protected static CodeBlock applyCustomResponseHandler(OperationModel operationModel,
```

Comment:
Would it make more sense to have this return an Optional<CodeBlock> and in the calling code do .orElseGet(protocolSpec::responseHandler)?